stl/inc/format

```diff
@@ -1168,22 +1175,22 @@ public:
         _Out _OutputIt_, basic_format_args<basic_format_context> _Ctx_args, const locale& _Loc_)
         : _OutputIt(_OutputIt_), _Args(_Ctx_args), _Loc(_Loc_) {}
 
-    basic_format_arg<basic_format_context> arg(size_t _Id) const {
+    _NODISCARD basic_format_arg<basic_format_context> arg(size_t _Id) const {
         return _Args.get(_Id);
     }
-    locale locale() {
+    _NODISCARD locale locale() {
         return _Loc;
     }
 
-    iterator out() {
+    _NODISCARD iterator out() {
         return _OutputIt;
     }
     void advance_to(iterator _It) {
         // TODO: IDL support probably required
         _OutputIt = _It;
     }
 
-    const basic_format_args<basic_format_context>& _Get_args() const {
+    _NODISCARD const basic_format_args<basic_format_context>& _Get_args() const {
         return _Args;
     }
 };

```

Comment:
`_Get_format_arg_type_storage_size` on 1199 should be `_NODISCARD` to agree with the forward declaration on 887.

Reply:
I always get confused where attributes have to go. Is this just to make them "agree" without affecting diag or is this necessary?

Reply:
Formally, this is just fine per [dcl.attr.nodiscard]/2 "A name or entity declared without the `nodiscard` attribute can later be redeclared with the attribute and vice-versa." If any declaration is `[[nodiscard]]`, the compiler should treat the function as `[[nodiscard]]`.

Conventionally, however, the STL tries to ensure that all attributes are the same on all declarations of a name to avoid confusion.